taggit/managers.py

```diff
@@ -203,9 +206,31 @@ def add(self, *tags):
         for new_tag in tags_to_create:
             tag_objs.add(self.through.tag_model().objects.create(name=new_tag))
 
+        new_ids = set(t.pk for t in tag_objs)
+
+        # NOTE: can we hardcode 'tag_id' here or should the column name be got
+        # dynamically from somewhere?
+        vals = (self.through._default_manager.using(db)
```

Comment:
This adds an additional SQL query, however, as the method already produces `O(n)` SQL queries it's rather semantic. There is potentially room for using `bulk_create` to achieve `O(1)`, however, that is outside the scope of this PR.